packages/core/src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -686,4 +693,120 @@ export class AbstractQueryInterfaceTypeScript {
   ): Promise<void> {
     await this.sequelize.queryRaw(this.queryGenerator.getToggleForeignKeyChecksQuery(enable), options);
   }
+
+  /**
+   * Create a new savepoint
+   *
+   * @param transaction
+   * @param options
+   */
+  async createSavepoint(transaction: Transaction, options: CreateSavepointOptions): Promise<void> {
+    if (!this.queryGenerator.dialect.supports.savepoints) {
+      throw new Error('Savepoints are not supported for the current dialect!');
+    }
+
+    if (!transaction || !(transaction instanceof Transaction)) {
+      throw new Error('Unable to create a savepoint without transaction object!');
+    }
+
+    const sql = this.queryGenerator.createSavepointQuery(options.savepointName);
+    await this.sequelize.queryRaw(sql, { ...options, transaction, supportsSearchPath: false });
+  }
+
+  /**
+   * Begin a new transaction
+   *
+   * @param transaction
+   * @param options
+   */
+  async startTransaction(transaction: Transaction, options: StartTransactionOptions): Promise<void> {
+    if (!transaction || !(transaction instanceof Transaction)) {
+      throw new Error('Unable to start a transaction without transaction object!');
+    }
+
+    const sql = this.queryGenerator.startTransactionQuery(options);
+    await this.sequelize.queryRaw(sql, { ...options, transaction, supportsSearchPath: false });
+  }
+
+  /**
+   * Commit an already started transaction
+   *
+   * @param transaction
+   * @param options
+   */
+  async commitTransaction(transaction: Transaction, options: CommitTransactionOptions): Promise<void> {
+    if (!transaction || !(transaction instanceof Transaction)) {
+      throw new Error('Unable to commit a transaction without transaction object!');
+    }
+
+    const sql = this.queryGenerator.commitTransactionQuery();
+    await this.sequelize.queryRaw(sql, {
+      ...options,
+      transaction,
+      supportsSearchPath: false,
+      completesTransaction: true,
```

Comment:
considering it's private, I think this should be an internal Symbol

Reply:
Sorry I'm not sure what you are wanting me to do here.

Reply:
Instead of having a string key called `completesTransaction`, I think it should be a symbol (`Symbol('completesTransaction')`) that we don't export outside of Sequelize

still not fully private, but makes it harder to use that property

Reply:
Got, I've update this now

Reply:
Sorry for the nit but could you opt for a symbol exported from some file (maybe from transaction.ts? up to you) instead of a globally registered symbol?

Reply:
No worries, updated now